mscore/musescore.cpp

```diff
@@ -2224,8 +2224,38 @@ static bool processNonGui()
                         }
                   }
             else if (fn.endsWith(".png")) {
-                  cs->switchToPageMode();
-                  rv = mscore->savePng(cs, fn);
+                  if (!exportScoreParts) {
+                        cs->switchToPageMode();
+                        return mscore->savePng(cs, fn);
+                        }
+                  else {
+                        if (cs->excerpts().size() == 0) {
+                              auto exceprts = Excerpt::createAllExcerpt(cs->masterScore());
+                      
+                              for(Excerpt* e: exceprts) {
+                                    Score* nscore = new Score(e->oscore());
+                                    e->setPartScore(nscore);
+                                    nscore->setName(e->title()); // needed before AddExcerpt
+                                    nscore->style()->set(StyleIdx::createMultiMeasureRests, true);
+                                    cs->startCmd();
+                                    cs->undo(new AddExcerpt(nscore));
+                                    createExcerpt(e);
+                                    cs->endCmd();
+                                    }
+                              }
+                        if (!mscore->savePng(cs, fn))
+                              return false;
+                        int idx = 0;
+                        int padding = QString("%1").arg(cs->excerpts().size()).size();
+                        for(Excerpt* e: cs->excerpts()) {
+                              QString suffix = QString("__excerpt__%1.png").arg(idx, padding, 10, QLatin1Char('0'));
```

Comment:
Sorry, but I did not see how would a PDF file be exported in its (part? instrument? track?) name of excerpt. All excerpts were exported into a single PDF file.